fl4health/PCA/pca.py

```diff
@@ -0,0 +1,62 @@
+from logging import INFO
+from pathlib import Path
+from typing import Tuple
+
+import numpy as np
+from flwr.common.logger import log
+from flwr.common.typing import NDArray, NDArrays
+from sklearn.decomposition import PCA
+
+
+class ClientSidePCA:
+    def __init__(self, n_components: int) -> None:
+        self._pca = PCA(n_components=n_components, random_state=42)
+        self.principal_components: NDArray
+        self.eigenvalues: NDArray
+
+    def compute_pc(self, X: NDArray) -> Tuple[NDArray, NDArray]:
+        self._pca.fit(X)
+        components, eigen_vals = self._pca.components_, self._pca.explained_variance_
+        return components, eigen_vals
+
+    def update_pcs(self, pcs: NDArray, eigenvalues: NDArray) -> None:
+        self.principal_components = pcs
+        self.eigenvalues = eigenvalues
+
+    def get_pcs(self) -> Tuple[NDArray, NDArray]:
+        return self.principal_components, self.eigenvalues
+
+    def pc_projection(self, X: NDArray) -> NDArray:
+        return self.principal_components @ X
+
+    def save_pcs(self, pc_path: Path) -> None:
+        pcs, eigenvalues = self.get_pcs()
+        log(INFO, f"principal components shape: {pcs.shape}")
+        log(INFO, f"eigenvalues shape: {eigenvalues.shape}")
+        np.save(pc_path, pcs)
+
+
```

Comment:
This class is not currently used since I changed design and implemented the merging in a strategy class instead. The alternative is to create a separate `FedPCAServer` class that leverages this to perform merging.